napari/_qt/dialogs/qt_plugin_dialog.py

```diff
@@ -620,20 +324,27 @@ def addItem(
         item.setSizeHint(widg.sizeHint())
         self.setItemWidget(item, widg)
 
-    def handle_action(self, item, pkg_name, action_name, update=False):
+    def handle_action(
+        self,
+        item: QListWidgetItem,
+        pkg_name: str,
+        action_name: str,
+        update: bool = False,
+    ):
+        # TODO: 'task_handler' should be configurable per item, depending on dropdown
+        task_handler = "conda" if running_as_constructor_app() else "pip"
         widget = item.widget
         item.setText(f"0-{item.text()}")
-        method = getattr(self.installer, action_name)
         self._remove_list.append((pkg_name, item))
         self._warn_dialog = None
-        if item.npe_version != 1:
+        # TODO: NPE version unknown before installing
+        if item.npe_version != 1 and action_name == "uninstall":
             # show warning pop up dialog
             message = trans._(
-                'When installing/uninstalling npe2 plugins, you must restart napari for UI changes to take effect.'
-            )
-            self._warn_dialog = WarnPopup(
-                text=message,
+                'When installing/uninstalling npe2 plugins, you must '
+                'restart napari for UI changes to take effect.'
             )
+            self._warn_dialog = WarnPopup(text=message)
```

Comment:
That's an excellent point. I don't like the current popup mechanism (it gets in the way; the user has to dismiss it _every time_ to recover focus). So maybe a more general message in the UI is enough? We can cover that in a different PR though, since this is getting quite large now.